feelpp/feel/feeldiscr/minmax.hpp

```diff
@@ -0,0 +1,181 @@
+ * @copyright Copyright (c) 2022 Université de Strasbourg
+ * 
+ */
+#pragma once
+
+#include <feel/feelalg/glas.hpp>
+#include <feel/feelcore/parameter.hpp>
+#include <feel/feelcore/serialization.hpp>
+#include <feel/feeldiscr/functionspace.hpp>
+
+namespace Feel {
+
+/**
+ * @brief compute the min of the element of a function space at DOF points over the range @p r
+ * 
+ * @tparam RangeT range type to iterate on
+ * @tparam ElementT type of finite element function
+ * 
+ * @param r range of element
+ * @param e element of a function space
+ * 
+ * @code {.cpp}
+ * auto v = Xh->element();
+ * auto [min_v,arg_min_v] = minelt(_range=elements(mesh),_element=v);
+ * @endcode
+ * 
+ * @return the min value and the dof point coordinates
+ */
+template <typename... Ts>
+auto minelt( Ts&&... v )
+{
+    auto args = NA::make_arguments( std::forward<Ts>( v )... );
+    auto && r = args.get(_range);
+    auto && e = args.get(_element);
+
+    using value_type = typename decay_type<decltype(e)>::value_type;
+    using functionspace_type = typename decay_type<decltype(e)>::functionspace_type;
+    constexpr int nRealDim = functionspace_type::nRealDim;
+    double min_v{std::numeric_limits<value_type>::max()};
+    int min_id{ 0 };
+
+    for ( auto const& rangeElt : r )
+    {
+        auto const& meshElt = boost::unwrap_ref( rangeElt );
+        index_type eid = meshElt.id();
+
+        for ( uint16_type local_id=0; local_id < functionspace_type::fe_type::nLocalDof; ++local_id )
+        {
+            int dofpn = e.functionSpace()->dof()->localToGlobal( eid, local_id, 0 ).index();
+            value_type d =e( dofpn );
+            if ( min_v > d )
+            {
+                min_v = d;
+                min_id = dofpn;
+            }
+        }
+    }
+    auto [pt,thedof,comp] = e.functionSpace()->dof()->dofPoint(min_id);
+    eigen_vector_type<nRealDim> e_pt = emap<value_type>( pt );
+    auto local_min = std::tuple{min_v,e_pt};
+    std::vector<std::tuple<value_type,eigen_vector_type<nRealDim>>> all_min(e.functionSpace()->worldComm().globalSize());
+    mpi::all_gather( e.functionSpace()->worldComm().globalComm(), local_min, all_min );
+    return *std::min_element( all_min.begin(), all_min.end(), 
+                                []( auto const& a, auto const& b  ){
+                                    return std::get<0>(a) < std::get<0>(b); }
+                             );
+}
+
+/**
+ * @brief compute the max of the element of a function space at DOF points over the range @p r
+ *
+ * @tparam RangeT range type to iterate on
+ * @tparam ElementT type of finite element function
+ *
+ * @param r range of element
+ * @param e element of a function space
+ *
+ * @code {.cpp}
+ * auto v = Xh->element();
+ * auto [max_v,arg_max_v] = maxelt(_range=elements(mesh),_element=v);
+ * @endcode
+ *
+ * @return the max value and the dof point coordinates
+ */
+template <typename... Ts>
+auto maxelt( Ts&&... v )
```

Comment:
this maxelt function is very close to the minelt, maybe can be factorized

Reply:
yes that's planned as well